action.yaml

```diff
@@ -0,0 +1,23 @@
+name: Charmhub Upload
+description: Uploads a charm to charmhub.io
+author: Kenneth Koski
+inputs:
+  credentials:
+    description: |
+      Credentials exported from `charmcraft login --export`. See
+      https://juju.is/docs/sdk/remote-env-auth for more info
+    required: true
+  charm-path:
+    description: Path to charm directory
+    required: false
+    default: "."
+  charmcraft-channel:
+    description: Snap channel to use when installing charmcraft
+    required: false
+    default: "latest/edge"
```

Comment:
Should probably default to `latest/stable` to make the default as reliable as possible.

Reply:
I suspect @knkski defaulted it to edge due to the charmcraft tokens part not being in stable yet. We will have to set it to edge in all the current actions for it to work correctly but `latest/stable` should be the default.

Reply:
> We will have to set it to edge in all the current actions for it to work correctly but latest/stable should be the default.

You only need edge to get the token **out** though, or did I miss something? 😅

Reply:
Thats possible, let me try it out.

Reply:
Just [tested](https://github.com/canonical/kubeflow-volumes-operator/runs/4380734897?check_suite_focus=true) it, and no, you need to use `latest/edge` for it to work correctly. I propose leaving it to `latest/edge` by default and opening an issue so that we move it to `latest/stable` once these features are released there.